overlord/snapstate/component.go

```diff
@@ -30,8 +31,189 @@ import (
 	"github.com/snapcore/snapd/overlord/state"
 	"github.com/snapcore/snapd/snap"
 	"github.com/snapcore/snapd/snap/naming"
+	"github.com/snapcore/snapd/store"
+	"github.com/snapcore/snapd/strutil"
 )
 
+// InstallComponents installs all of the components in the given names list. The
+// snap represented by info must already be installed, and all of the components
+// in names should not be installed prior to calling this function.
+//
+// TODO:COMPS: respect the transaction that is passed to this function
+func InstallComponents(ctx context.Context, st *state.State, names []string, info *snap.Info, opts Options) ([]*state.TaskSet, error) {
```

Comment:
we need conflict tetsts for this, that this conflict with other refresh operations on the snap? that this conflict with itself, and this when set up conflicts with starting a new snap operation

Reply:
Added tests here: https://github.com/canonical/snapd/pull/14260/commits/0d6b7fe00664f56841a2880363034276eaab36c9